src/plugins/data/public/query/saved_query/saved_query_service.test.ts

```diff
@@ -287,6 +292,175 @@ describe('saved query service', () => {
+      });
+
+      const response = await getSavedQuery('foo');
+      expect(response.attributes.query.query).toEqual({
+        email: [
+          { type: 'work', address: 'work@***.com' },
+          { type: 'home', address: 'home@***.com' },
+        ],
+      });
+    });
+
+    it('should correctly parse a json object', async () => {
+      mockSavedObjectsClient.get.mockReturnValue({
+        id: 'foo',
+        attributes: {
+          title: 'foo',
+          description: 'bar',
+          query: {
+            language: 'kuery',
+            query: {
+              email: [
+                { type: 'work', address: 'work@***.com' },
+                { type: 'home', address: 'home@***.com' },
+              ],
+            },
+          },
+        },
+      });
+
+      const response = await getSavedQuery('foo');
+      expect(response.attributes.query.query).toEqual({
+        email: [
+          { type: 'work', address: 'work@***.com' },
+          { type: 'home', address: 'home@***.com' },
+        ],
+      });
+    });
+
+    it('should handle null string with single quote', async () => {
+      mockSavedObjectsClient.get.mockReturnValue({
+        id: 'foo',
+        attributes: {
+          title: 'foo',
+          description: 'bar',
+          query: {
+            language: 'kuery',
+            query: 'null',
+          },
+        },
+      });
+
+      const response = await getSavedQuery('foo');
+      expect(response.attributes.query.query).toEqual('null');
+    });
+
+    it('should handle null string with double quote', async () => {
+      mockSavedObjectsClient.get.mockReturnValue({
+        id: 'foo',
+        attributes: {
+          title: 'foo',
+          description: 'bar',
+          query: {
+            language: 'kuery',
+            query: 'null',
+          },
+        },
+      });
+
+      const response = await getSavedQuery('foo');
+      expect(response.attributes.query.query).toEqual('null');
+    });
+
+    it('should handle null quoted string', async () => {
+      mockSavedObjectsClient.get.mockReturnValue({
+        id: 'foo',
+        attributes: {
+          title: 'foo',
+          description: 'bar',
+          query: {
+            language: 'kuery',
+            query: '"null"',
+          },
+        },
+      });
+
+      const response = await getSavedQuery('foo');
+      expect(response.attributes.query.query).toEqual('"null"');
+    });
+
+    it('should handle undefined string with single quote', async () => {
+      mockSavedObjectsClient.get.mockReturnValue({
+        id: 'foo',
+        attributes: {
+          title: 'foo',
+          description: 'bar',
+          query: {
+            language: 'kuery',
+            query: 'undefined',
+          },
+        },
+      });
+
+      const response = await getSavedQuery('foo');
+      expect(response.attributes.query.query).toEqual('undefined');
+    });
+
+    it('should handle undefined string with double quote', async () => {
```

Comment:
This test case seems wrong, how is it different from the previous test case?

Reply:
Will remove all unit tests on undefined.